drivers/usb/device/usb_dc_numaker.c

```diff
@@ -0,0 +1,1943 @@
+			    &usb_dc_numaker_device_##inst, inst);                                  \
+                                                                                                   \
+		irq_enable(DT_INST_IRQN(inst));                                                    \
+	}                                                                                          \
+                                                                                                   \
+	static void usb_dc_numaker_irq_unconfig_func_##inst(struct usb_dc_numaker_device *dev)     \
+	{                                                                                          \
+		irq_disable(DT_INST_IRQN(inst));                                                   \
+	}                                                                                          \
+                                                                                                   \
+	static const struct usb_dc_numaker_config usb_dc_numaker_config_##inst = {                 \
+		.usbd_base = (USBD_T *)DT_INST_REG_ADDR(inst),                                     \
+		.reset = RESET_DT_SPEC_INST_GET(inst),                                             \
+		.clk_modidx = DT_INST_CLOCKS_CELL(inst, clock_module_index),                       \
+		.clk_src = DT_INST_CLOCKS_CELL(inst, clock_source),                                \
+		.clk_div = DT_INST_CLOCKS_CELL(inst, clock_divider),                               \
+		.clkctrl_dev = DEVICE_DT_GET(DT_PARENT(DT_INST_CLOCKS_CTLR(inst))),                \
+		.irq_config_func = usb_dc_numaker_irq_config_func_##inst,                          \
+		.irq_unconfig_func = usb_dc_numaker_irq_unconfig_func_##inst,                      \
+		.pincfg = PINCTRL_DT_INST_DEV_CONFIG_GET(inst),                                    \
+		.num_bidir_endpoints = DT_INST_PROP(inst, num_bidir_endpoints),                    \
+		.dmabuf_size = DT_INST_PROP(inst, dma_buffer_size),                                \
+		.maximum_speed =                                                                   \
+			DT_INST_ENUM_IDX_OR(inst, maximum_speed, DT_USB_MAXIMUM_SPEED_FULL_SPEED), \
+	};                                                                                         \
+                                                                                                   \
+	static struct usb_dc_numaker_data usb_dc_numaker_data_##inst;                              \
+                                                                                                   \
+	BUILD_ASSERT(DT_INST_PROP(inst, num_bidir_endpoints) == USBD_MAX_EP,                       \
+		     "num_bidir_endpoints doesn't match BSP USBD driver");                         \
+                                                                                                   \
+	static struct usb_dc_numaker_device usb_dc_numaker_device_##inst = {                       \
```

Comment:
NACK for usb_dc_numaker_device and SYS_INIT, use 

```
DEVICE_DT_INST_DEFINE(n, usb_dc_numaker_init, NULL,
    &usb_dc_numaker_data_##n, &usb_dc_numaker_config_##n, ...
```

Reply:
replaced `SYS_INIT` with `DEVICE_DT_INST_DEFINE`